seq/alphabet_conv.go

```diff
@@ -521,118 +521,33 @@ func RNAToProtein(rna []byte, transl_table int) (protein []byte, err error) {
-				}
-			case 9:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable9[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 10:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable10[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 11:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable11[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 12:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable12[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 13:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable13[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 14:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable14[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 16:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable16[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 21:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable21[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 22:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable22[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 23:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable23[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 24:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable24[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 25:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable25[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 26:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable26[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 27:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable27[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 28:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable28[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 29:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable29[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 30:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable30[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
-			case 31:
-				if aminoacid, ok := rnaCodonToAminoAcidTranslTable31[last3NTs.String()]; ok {
-					protein = append(protein, []byte(aminoacid)...)
-				}
+	var protein bytes.Buffer
+	for i := 0; i < len(dna)-2; i += 3 {
+		if aminoacid, ok := condonTables[transl_table][string(dna[i:i+3])]; ok {
```

Comment:
optimizations:

- assigning `condonTables[transl_table]` to a variable can reduce memory access time.
- make  `ok` a variable outside `for` loop.